File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;
@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;
    
	@RequestMapping("/findAll")
    public List<TbBrand> findAll(){
    return 	brandService.findAll();
    }
	
	
	@RequestMapping("/findPage")
	public PageResult findPage(int page,int size) {
	return brandService.findPage(page, size);
	}
	
	
	@RequestMapping("/addBrand")
	public Result saveBrand(@RequestBody TbBrand tbBrand) {
		try {
			brandService.save(tbBrand);
			return new Result(true,"保存成功");
		} catch (Exception e) {
			// TODO Auto-generated catch block
			return new Result(false, "保存失败");
		}
		
	}
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand tbBrand) {
		
		try {
			brandService.update(tbBrand);
			return new Result(true,"修改成功");
		} catch (Exception e) {
			// TODO Auto-generated catch block
			return new Result(false, "修改失败");
		}
		
	}
	@RequestMapping("/findOne")
	public TbBrand findOne(long id) {
		return brandService.findOne(id);
	}
	@RequestMapping("/delete")
	public Result delete(Long [] ids) {
		try {
			brandService.delete(ids);
			return new Result(true,"删除成功");
		} catch (Exception e) {
			// TODO Auto-generated catch block
			return new Result(false, "删除失败");
		}
	}
	
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand tbBrand,int page,int size) {
		return brandService.findPage(tbBrand,page,size);

	}
	
	@RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
		return brandService.selectOptionList();
	}	
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
 app.controller('brandController',function($scope,$controller,brandService){
	       $controller('baseController',{$scope:$scope});
    	   $scope.findAll=function(){
    		   brandService.findAll().success(function(response){
    			   $scope.brands=response;
    		   });
    	   }
    		 
    	 //分页
    	    $scope.findPage=function(page,rows){
    	    	brandService.findPage(page,rows).success(function(response){
    	    		$scope.brands=response.rows;  //显示数据
    	    		$scope.paginationConf.totalItems=response.total; //将总记录数赋值给paginationConf
    	    	});
    	    }
    	 //新增
    	 $scope.save=function(){
    		 var object=brandService.add($scope.b);
    		 if($scope.b.id!=null){
    			 object=brandService.update($scope.b);
    		 }
    		 object.success(function(rp){
    			 if(rp.success){
    				 $scope.reloadList();
    			 }else{
    				 alart(rp.message);
    			 }
    		 });
    	 }
    	
    	 //查找对象
    	 $scope.findOne=function(id){
    		 brandService.findOne(id).success(function(rp){
    			 $scope.b=rp;
    		 });
    	 }
    	  
    	
    	 //删除选中
    	 $scope.dele=function(){
    		 brandService.dele($scope.selectIds).success(function(rp){
    			 if(rp.success){
    				 $scope.reloadList();
    			 }else{
    				 alart(rp.message);
    			 }
    		 });
    	 }
    	 //搜索
    	 $scope.search=function(page,size){
    		    brandService.search(page,size,$scope.s).success(function(response){
 	    		$scope.brands=response.rows;  //显示数据
 	    		$scope.paginationConf.totalItems=response.total; //将总记录数赋值给paginationConf
 	    	});
    	 }
       });